unblob/models.py

```diff
@@ -131,6 +134,25 @@ def update(self, other: "ProcessResult"):
     def register(self, result: TaskResult):
         self.results[result.task] = result
 
+    def to_json(self, indent="  "):
+        return json.dumps(list(self.results.values()), cls=_JSONEncoder, indent=indent)
+
+
+class _JSONEncoder(json.JSONEncoder):
```

Comment:
I know, we had discussed IRL about using `attr.asdict`-s features to do the type conversion. I don't recall exactly why we didn't go down that route. If we could create a Python representation of the json, which can be converted to string without a custom encoder would help testing the feature tremendously.

Reply:
I remember you said something about it, but most probably the channel was bad/noisy and I did not understand your proposal.